### PR TITLE
Add support for vina to prolif interaction reward

### PR DESCRIPTION
**Pull request by tyszw**
@sishida21 
I added an example of a reward function to evaluate interactions using Vina and ProLIF.
Please confirm 4b11972c11dd95d3c4975f7c5d3dfda011221713 for details.

### CODE REVIEW

File: reward/util.py [hunk 1]
```diff
@@ -509,18 +510,36 @@ def calc_strain_energy(docking_pose_file, conf):
...
 
     # Import necessary modules
     plf = sys.modules.get('prolif') or __import__('prolif')
     mda = sys.modules.get('MDAnalysis') or __import__('MDAnalysis')
 
-    # Load receptor and docked ligand
-    protein_file = receptor_fname.replace('/scr', 'data')
-    u = mda.Universe(protein_file)
+    # Load receptor
+    if receptor_fname.startswith('/scr'):
```
Comment (sishida21) [inline reward/util.py:520]: Please add a comment: "This process is for cases where GNINA reward is used".

File: reward/util.py [hunk 2]
```diff
@@ -509,18 +510,36 @@ def calc_strain_energy(docking_pose_file, conf):
...
+            string = f.read()
+        pdbqt_mol = meeko.PDBQTMolecule(string, is_dlg=False, skip_typing=True)
+        sdf_content = meeko.RDKitMolCreate.write_sd_string(pdbqt_mol, only_cluster_leads=False)[0]
+        with tempfile.NamedTemporaryFile(mode='w', suffix='.sdf', delete=False) as tf:
+            tf.write(sdf_content)
+            tf_path = tf.name
+        pose_iterable = plf.sdf_supplier(tf.name)
+        os.remove(tf_path)
+    else:
+        raise ValueError(f"Unsupported file format: {output_ligand_ext}")
```
Comment (sishida21) [inline reward/util.py:542]: Does this modification result in an error when running GNINA reward? I think that GNINA outputs ligand files with sdf extension, which may raise ValueError in this process.

**Review by sishida21 — COMMENTED** [review]
Please check the above comments.

File: reward/util.py (hunk 1 above)
Comment (tyszw) [reply reward/util.py]: I added the comment. (b3584684ede807fea7024d270e448f69981378ef)

File: reward/util.py (hunk 2 above)
Comment (tyszw) [reply reward/util.py]: I’ve fixed it to avoid ValueError. (38b1f9d66be9751fd1e5fe17e32e471efe7e2a5a)

**Comment by tyszw** [comment]
@sishida21 
Thank you for your review. I have made the changes based on your feedback. Please let me know if you have any other concerns.

**Pull request merged**